pkg/skaffold/docker/image_test.go

```diff
@@ -187,9 +188,11 @@ func TestGetBuildArgs(t *testing.T) {
 				BuildArgs: map[string]*string{
 					"key1": util.StringPtr("value1"),
 					"key2": nil,
+					"key3": util.StringPtr("{{.FOO}}"),
 				},
 			},
-			want: []string{"--build-arg", "key1=value1", "--build-arg", "key2"},
+			env:  []string{"FOO=bar"},
```

Comment:
Please add a test when Env Variable is not set and the BuildArgs should fail.

You can do this by adding a new boolean field in the test struct
```
		description string
		artifact    *latest.DockerArtifact
		env         []string
		want        []string
+              shouldErr bool
```
Add a test case where 
```
description: "build args",
			artifact: &latest.DockerArtifact{
				BuildArgs: map[string]*string{
					"key1": util.StringPtr("value1"),
					"key2": nil,
					"key3": util.StringPtr("{{.DOES_NOT_EXIST}}"),
				},
			},
                       shouldErr : true
           }
```
And test if GetBuildArgs sees an error here
```
  result, err := GetBuildArgs(tt.artifact)
  if test.shouldErr && err != nil {
   t.Errorf("expected to see an error, but saw none")
  } 
```